src/connection.js

```diff
@@ -25,16 +24,15 @@ module.exports = function connection (swarm) {
       swarm.handle(muxer.multicodec, (protocol, conn) => {
         const muxedConn = muxer.listener(conn)
 
-        muxedConn.on('stream', (conn) => {
-          protocolMuxer(swarm.protocols, conn)
-        })
+        muxedConn.on('stream', swarm.protocolMuxer(protocol))
```

Comment:
my bad for not renaming things here. Mind s/swarm/switch/g all the occurrences?

Reply:
I would, but `switch` is a reserved JS keyword.. ;)
Switching to `swtch`, I lack of a better replacement…